### PR TITLE
feat: try flutter_launcher_icons.yaml config file first and allow overriding configFile

### PR DESCRIPTION
**Pull request by alextekartik**
Hi and thanks for the plugin. For you information I use this tool not as a dependency but as an activated package
```
pub global activate flutter_launcher_icons
```
And run it this way
```
pub global run flutter_launcher_icons:main
```
i.e. I like it better not to have my to have an extra dependency (especially since this package has a git dependency) on my flutter app.

My main issue is that I don't like putting too many settings in pubspec.yaml which is typically reserved for...pub and build. Changing this configuration does not affect build, it is rather a one time tool that you run on demand but not before each build.
So my proposition is to make `flutter_launcher_icons.yaml` the default config file (if not found it will try `pubspec.yaml`) and to allow overriding this file.

So doing this
```
pub global run flutter_launcher_icons:main
```
will try `flutter_launcher_icons.yaml` then `pubspec.yaml`

doing this
```
pub global run flutter_launcher_icons:main -f my_icons.yaml
```
will try `my_icons.yaml` only

### CODE REVIEW

**Comment by MarkOSullivan94** [comment]
Thanks for the PR!

Currently working on #33 so once that's finished, I'll merge this in and check it's working as expected. Appreciate you adding in a unit test to ensure it wont be broken in the future!

**Comment by MarkOSullivan94** [comment]
Only getting round to checking this out now. I'm wanting to get this merged in before doing another release. Got a couple of new features and fixes in the 0.7 release and it's been refactored quite a bit.

Would you be able to update your branch with a working example which I can test out? 

I was testing it out and it worked fine whenever I had added a flutter_launcher_icons.yaml to the example project (using a similar example to what you included in the test directory) however it didn't work whenever I commented out everything for Flutter Launcher Icons within the pubspec.yaml file.

Here's my flutter_launcher_icons.yaml

```
dev_dependencies:

  flutter_launcher_icons:
    path: ../../

flutter_icons:
  image_path: "assets/images/icon-710x599.png"
  android: true # can specify file name here e.g. "ic_launcher"
  ios: true # can specify file name here e.g. "My-Launcher-Icon"
```

terminal log
```
mark@marks-pc:example acf295b ✗ 6d △ ✓ ⍉ ➜ pub global run flutter_launcher_icons:main -f flutter_launcher_icons.yaml
Unhandled exception:
NoSuchMethodError: The method 'containsKey' was called on null.
Receiver: null
Tried calling: containsKey("image_path")
#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:50:5)
#1      isImagePathInConfig (package:flutter_launcher_icons/main.dart:54:31)
#2      createIcons.<anonymous closure> (package:flutter_launcher_icons/main.dart:10:10)
#3      _RootZone.runUnary (dart:async/zone.dart:1379:54)
#4      _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#5      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
#6      Future._propagateToListeners (dart:async/future_impl.dart:671:32)
#7      Future._complete (dart:async/future_impl.dart:476:7)
#8      _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#9      _AsyncAwaitCompleter.complete (dart:async/runtime/libasync_patch.dart:28:18)
#10     _completeOnAsyncReturn (dart:async/runtime/libasync_patch.dart:295:13)
#11     loadConfigFile (package:flutter_launcher_icons/main.dart)
<asynchronous suspension>
#12     createIcons (package:flutter_launcher_icons/main.dart:8:3)
<asynchronous suspension>
#13     main (file:///Users/mos/.pub-cache/hosted/pub.dartlang.org/flutter_launcher_icons-0.6.1/bin/main.dart:5:3)
#14     _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:287:32)
#15     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```

Note that example shows me forcing it to use the flutter_launcher_icons.yaml file just in case it was just the default behaviour which was having an issue.

**Comment by alextekartik** [comment]
Sorry that you get troubles. I added more tests. My guess is that you are still using the old executable. When you use `pub global run` have you activated it using the `path` option (otherwise it is still using the pub version). Can you try using `dart bin/main.dart` instead until it is published?

You can try the `--help` option to see if you are using the proper version, it should give:
```
Generates icons for iOS and Android
-h, --help    Usage help
-f, --file    Config file (default: flutter_launcher_icons.yaml)
```

**Comment by MarkOSullivan94** [comment]
I didn't activate it using the path, my mistake! Thanks for the tip. 

Was able to run it successfully using the flutter_laucher_icons.yaml config with no file specified and then using the -f flag I was able to run it again specifying for it to look for the renamed config file which I changed to `test_config.yaml`.

Really like these changes, great work! Love the help flag being included too. I might build upon that once this is merged in.

I did notice that if there's no config found that it throws this error:

```
NoSuchMethodError: The method 'containsKey' was called on null.
Receiver: null
Tried calling: containsKey("image_path")
```

I think it'd be better to throw a more human readable exception, something like `NoConfigFoundException` which you could add to the `customs_exception.dart` file.

After that, I think it'll be ready to be merged in when the merge conflict is tackled.

**Comment by alextekartik** [comment]
Added (well the exception is not thrown but displayed if the file is not found or if there is no flutter_icons section). Indeed I did not change this part before.

File: lib/main.dart [hunk 1]
```diff
@@ -28,7 +28,9 @@ createIconsFromArguments(List<String> arguments) async {
   // Load the config file
   var yamlConfig =
       await loadConfigFileFromArgResults(argResults, verbose: true);
-  if (yamlConfig == null) {
+  if ((yamlConfig == null) || (!(yamlConfig["flutter_icons"] is Map))) {
```
Comment (MarkOSullivan94) [inline lib/main.dart]: Great work! Thanks for this change

**Pull request merged**